amr-wind/equation_systems/icns/source_terms/ABLMesoForcingMom.cpp

```diff
@@ -53,29 +57,21 @@ void ABLMesoForcingMom::mean_velocity_init(const ABLMesoscaleInput& ncfile)
 void ABLMesoForcingMom::mean_velocity_init(
     const VelPlaneAveragingFine& vavg, const ABLMesoscaleInput& ncfile)
 {
+    const int num_meso_ht = ncfile.nheights();
+    m_nht = vavg.ncell_line();
     m_axis = vavg.axis();
     // The implementation depends the assumption that the ABL statistics class
     // computes statistics at the cell-centeres only on level 0. If this
     // assumption changes in future, the implementation will break... so put in
     // a check here to catch this.
-    AMREX_ALWAYS_ASSERT(
-        m_mesh.Geom(0).Domain().length(m_axis) ==
-        static_cast<int>(vavg.line_centroids().size()));
+    AMREX_ALWAYS_ASSERT(m_mesh.Geom(0).Domain().length(m_axis) == m_nht);
 
-    m_nht = static_cast<int>(vavg.line_centroids().size());
     m_zht.resize(m_nht);
-
-    m_velAvg_ht.resize(vavg.line_centroids().size());
-    m_uAvg_vals.resize(vavg.ncell_line());
-    m_vAvg_vals.resize(vavg.ncell_line());
-
-    m_meso_avg_error.resize(vavg.ncell_line());
-
-    m_error_meso_avg_U.resize(vavg.ncell_line());
-    m_error_meso_avg_V.resize(vavg.ncell_line());
-
-    m_err_U.resize(vavg.ncell_line());
-    m_err_V.resize(vavg.ncell_line());
+    m_velAvg_ht.resize(m_nht);
```

Comment:
silly comment but I will be taking care of this at some point. Can you make this a snake case variable name?